src/utils/signMessage.ts

```diff
@@ -0,0 +1,91 @@
+import type { SignTypedDataArgs } from '@wagmi/core';
+import { isAddress } from 'viem';
+
+/**
+ * Generate the TrustCredential typed data payload which will be signed by the user.
+ *
+ * @param issueAddress - The issuer address to be included in the message.
+ * @param subjectAddress - The subject address to be included in the message.
+ * @param chainId - The chainId to be included in the message.
+ * @param isAdd - The boolean to flag whether is for add or remove.
+ * @returns The message to be signed.
+ */
+export function generateTCTypedSignPayload(
+  issueAddress: string,
+  subjectAddress: string,
+  chainId: number,
+  isAdd: boolean,
+): SignTypedDataArgs {
+  if (!isAddress(issueAddress)) {
+    throw new Error('Invalid issueAddress');
+  }
+
+  if (!isAddress(subjectAddress)) {
+    throw new Error('Invalid subjectAddress');
+  }
+
+  const domain = {
+    name: 'EIP712TrustCredential',
+    version: '1',
+    chainId,
+  };
+
+  const types = {
+    Trustworthiness: [
+      {
+        name: 'scope',
+        type: 'string',
+      },
+      {
+        name: 'level',
+        type: 'uint8',
+      },
+      {
+        name: 'reason',
+        type: 'string[]',
+      },
+    ],
+    CredentialSubject: [
+      {
+        name: 'id',
+        type: 'string',
+      },
+      {
+        name: 'trustworthiness',
+        type: 'Trustworthiness[]',
+      },
+    ],
+
+    TrustCredential: [
+      {
+        name: 'type',
+        type: 'string[]',
+      },
+      {
+        name: 'issuer',
+        type: 'string',
+      },
+      {
+        name: 'credentialSubject',
+        type: 'CredentialSubject',
+      },
+    ],
+  };
+
+  const message = {
+    type: ['TrustCredential'],
+    issuer: `did:pkh:eip155:${chainId}:${issueAddress}`,
```

Comment:
TODO: 
add miss property

```
{
  "@context": [
    "https://www.w3.org/ns/credentials/v2",
    "https://www.w3.org/ns/credentials/examples/v2"
  ],
  "id": "http://university.example/credentials/3732",
  "type": ["VerifiableCredential"],
  "issuer": "my did",
  "validFrom": "2010-01-01T00:00:00Z",
  "credentialSubject": {
    "id": "destnation did"
  }
}
```

Reply:
validForm will be skipped because it is optional, type will be [`VerifiableCredential`, `AccountTrustCredential`]